kernel/version.c

```diff
@@ -19,3 +20,24 @@ uint32_t sys_kernel_version_get(void)
 {
 	return KERNELVERSION;
 }
+
+#ifdef CONFIG_LINKER_GNU_BUILD_ID
+
+/* Structure of .elf notes */
+struct elf_note_section {
+	uint32_t name_sz;
+	uint32_t desc_sz;
+	uint32_t type;
+	uint8_t data[];
+};
+
+/* Variable instantiated by the linker */
+extern const struct elf_note_section __g_note_build_id;
+
+const uint8_t *sys_gnu_build_id_get(void)
```

Comment:
In the event that this expands out into supporting other toolchains, the build ID may be differently sized... I'd suggest an interface more like this.

```suggestion
void sys_build_id_get(const uint8_t **buf, size_t *len)
```